sopel/cli/run.py

```diff
@@ -0,0 +1,619 @@
+        stderr('Sopel is not running!')
+        return ERR_CODE
+
+    stderr('Asking Sopel to restart')
+    if hasattr(signal, 'SIGUSR2'):
+        os.kill(pid, signal.SIGUSR2)
+    else:
+        # Windows will not generate SIGILL itself
+        # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+        os.kill(pid, signal.SIGILL)
+
+
+def command_legacy(opts):
+    """Legacy Sopel run script
+
+    The ``legacy`` command manages the old-style ``sopel`` command line tool.
+    Most of its features are replaced by the following commands:
+
+    * ``sopel start`` replaces the default behavior (run the bot)
+    * ``sopel stop`` replaces the ``--quit/--kill`` options
+    * ``sopel restart`` replaces the ``--restart`` option
+    * ``sopel configure`` replaces the
+      ``-w/--configure-all/--configure-modules`` options
+
+    The ``-v`` option for "version" is deprecated, ``-V/--version`` should be
+    used instead.
+
+    .. seealso::
+
+       The github issue `#1471`__ tracks various changes requested for future
+       version of Sopel, some of them are related to this legacy command line
+       tool.
+
+       .. __: https://github.com/sopel-irc/sopel/issues/1471
+
+    """
+    # Step One: Handle "No config needed" options
+    if opts.version:
+        print_version()
+        return
+    elif opts.version_legacy:
+        tools.stderr(
+            'option -v is deprecated, '
+            'use `sopel -V/--version` instead')
+        print_version()
+        return
+
+    if opts.wizard:
+        tools.stderr(
+            'option -w/--configure-all is deprecated, '
+            'use `sopel configure` instead')
+        _wizard('all', opts.config)
+        return
+
+    if opts.mod_wizard:
+        tools.stderr(
+            'option --configure-modules is deprecated, '
+            'use `sopel configure --modules` instead')
+        _wizard('mod', opts.config)
+        return
+
+    if opts.list_configs:
+        print_config()
+        return
+
+    # Step Two: Get the configuration file and prepare to run
+    try:
+        config_module = get_configuration(opts)
+    except ConfigurationError as e:
+        stderr(e)
+        return ERR_CODE_NO_RESTART
+
+    if config_module.core.not_configured:
+        stderr('Bot is not configured, can\'t start')
+        return ERR_CODE_NO_RESTART
+
+    # Step Three: Manage logfile, stdout and stderr
+    utils.redirect_outputs(config_module, opts.quiet)
+
+    # Step Four: Handle process-lifecycle options and manage the PID file
+    pid_dir = config_module.core.pid_dir
+    pid_file_path = get_pid_filename(opts, pid_dir)
+    old_pid = get_running_pid(pid_file_path)
+
+    if old_pid is not None and tools.check_pid(old_pid):
+        if not opts.quit and not opts.kill and not opts.restart:
+            stderr('There\'s already a Sopel instance running with this config file')
+            stderr('Try using either the `sopel stop` command or the `--restart` option')
```

Comment:
There's a `sopel restart` command now. Suggest that instead of `--restart`?

```suggestion
            stderr('Try using the `sopel stop` or `sopel restart` command')
```